### PR TITLE
refactor(core): Remove optimism e2e tests now that OVM 2.0 resembles EVM

### PR DESCRIPTION
**Pull request by nicholaspai**
**Motivation**

OVM 1.0 was significantly different from EVM so we created an e2e test to test contracts that were specifically compiled and deployed to it. OVM 2.0 resembles the EVM almost identically and we don't need e2e tests for it.

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [X]  All existing tests pass
- [ ]  Untested

### CODE REVIEW

File: packages/core/package.json [hunk 1]
```diff
@@ -14,9 +14,7 @@
   "devDependencies": {
     "@awaitjs/express": "^0.3.0",
     "@eth-optimism/contracts": "^0.4.5",
-    "@eth-optimism/message-relayer": "^0.1.8",
```
Comment (chrismaree) [inline packages/core/package.json:17]: the optimism bot (still wip) that transfers messages cross-chain will still need this package.

Comment (nicholaspai) [reply packages/core/package.json:17]: Re-added!

**Comment by mrice32** [comment]
@chrismaree @nicholaspai can we get this merged?

**Review by mrice32 — APPROVED** [review]

**Review by chrismaree — APPROVED** [review]
👋

**Pull request merged**